addons/account_unalterable/models/account_move.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, fields, models, _
+
+
+class AccountMove(models.Model):
+    _name = 'account.move'
+    _inherit = ['account.move', 'unalterable.hash.mixin']
+
+    @api.model
+    def _get_unalterable_fields(self):
+        # Override
+        return ['state', 'date', 'journal_id', 'company_id']
```

Comment:
IF we set this on the company and/or a db setting, then this will be available to configure in the saas (I think that should be the way)